EIPS/eip-4445.md

```diff
@@ -0,0 +1,179 @@
+
+### Name
+
+The chosen name resonates with the purpose of its existence. Consumers can be considered entities that utilise the token
+instances, without necessarily having ownership rights to it.
+
+The other name for the role that was considered was `operator`, however it is already defined and used within
+the `ERC-721` standard.
+
+### Restriction on the Permissions
+
+There are numerous use-cases where a distinct role for NFTs is required that MUST NOT have owner permissions. A contract
+that implements the consumer role and grants ownership permissions to the consumer renders this standard pointless.
+
+## Backwards Compatibility
+
+There are no other standards that define a similar role for NFTs and the name (`consumer`) is not used by other ERC-721
+related standards.
+
+## Reference Implementation
+
+The following is a snippet for reference implementation of the `ERC721Consumer` extension. The full repository can be
+found [here](https://github.com/Daniel-K-Ivanov/eip-721-consumer-extension)
+
+```solidity
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "./IERC721Consumable.sol";
+
+contract ERC721Consumable is IERC721Consumable, ERC721 {
+
+    // Mapping from token ID to consumer address
+    mapping (uint256 => address) _tokenConsumers;
+
+    constructor() ERC721("ReferenceImpl", "RIMPL") {
+    }
+
+    /**
+     * @dev See {IERC721Consumer-consumerOf}
+     */
+    function consumerOf(uint256 _tokenId) view external returns (address) {
+        require(_exists(_tokenId), "ERC721Consumer: consumer query for nonexistent token");
+        return _tokenConsumers[_tokenId];
+    }
+
+    /**
+     * @dev See {IERC721Consumer-changeConsumer}
+     */
+    function changeConsumer(address _consumer, uint256 _tokenId) external {
+        address owner = this.ownerOf(_tokenId);
+        require(msg.sender == owner || isApprovedForAll(owner, msg.sender),
+            "ERC721Consumer: changeConsumer caller is not owner nor approved for all");
+        _changeConsumer(owner, _consumer, _tokenId);
+    }
+
+    /**
+     * @dev Changes the consumer
+     * Requirement: `tokenId` must exist
+     */
+    function _changeConsumer(address _owner, address _consumer, uint256 _tokenId) internal {
+        _tokenConsumers[_tokenId] = _consumer;
+        emit ConsumerChanged(_owner, _consumer, _tokenId);
+    }
+
+    /**
+     * @dev See {IERC165-supportsInterface}.
+     */
+    function supportsInterface(bytes4 interfaceId) public view virtual override(IERC165, ERC721) returns (bool) {
+        return interfaceId == type(IERC721Consumable).interfaceId || super.supportsInterface(interfaceId);
+    }
+
+    function _beforeTokenTransfer(address _from, address _to, uint256 _tokenId) internal virtual override (ERC721) {
+        super._beforeTokenTransfer(_from, _to, _tokenId);
+
+        _changeConsumer(_from, address(0), _tokenId);
+    }
+
+}
+```
+
```

Comment:
There's also a "Security Considerations" section that needs to go here. It's ok if you say that there aren't any known considerations afaik, but I'd recommend talking about the potential cases where the permissions for the `consumer` aren't thought through clearly. (At least to me it does seem like there is a potential vector there - even if the letter of the standard is followed and the `consumer` can't transfer or burn the NFT, maybe they would otherwise be able to trigger things the owner wouldn't want.) This isn't a criticism - it's more of a warning to those implementing the standard to think things through.

Reply:
Good catch. Thank you for the feedback